Source/Core/InputCommon/ControllerEmu/StickGate.cpp

```diff
@@ -48,6 +48,42 @@ std::optional<double> GetRayLineIntersection(Common::DVec2 ray, Common::DVec2 po
   return diff.Cross(-point1) / dot;
 }
 
+std::tuple<double, double> ApplyVirtualNotches(double x, double y, double virtual_notch_degrees)
```

Comment:
This function would be incredibly simpler if you passed the angle and adjusted it instead of the x and y.
e.g. https://pastebin.com/4N9Frueh

You should adjust the `angle` value in `Reshape` after calculating `input_max_dist` and before calculating `gate_max_dist` to not break reshaping.
Reshaping is slightly broken in your PR (noticeable when very near the notch tip).

Rather than jumping to the exact notch angle, it might make sense to `MathUtil::Lerp` it towards the notch angle based on distance from center. (Pass the distance into `ApplyVirtualNotches`).
The notches wouldn't affect anything on a real controller until you hit the edge anyways. Also this would maintain smooth stick input.

Reply:
So I don't fully understand exactly what you mean here but what I can gather is that Lerp would kind of "nudge" the angle towards the notch using the midpoint? I don't quite like this concept since the point of the notches would be to force an exact angle on the joystick.

Having said that I was thinking since yesterday if we want this to actually behave a bit closer to notches it should really be inactive for most of the inner section of the stick. Maybe the virtual notch "zones" should look something more like the image below (thanks to my awesome photoshop skills):
![image](https://user-images.githubusercontent.com/9058133/97432784-6d604c80-1914-11eb-9624-d11e8a4a1146.png)

Essentially using the virtual notch size as a diameter for the circular sticky zones. Thoughts?

Reply:
So I added a new commit with a version of the above. Some of the logic feels a little bit hacky and there's a couple of other small things I don't like (e.g. should I include the `MINIMUM_NOTCH_DISTANCE` inside MappingIndicator when calculating the polygon points?) but I don't feel like I have enough of an understanding to improve it correctly. Take a look at eb8443234 and let me know what you think. Here's a GIF of the new UI and zones:

![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/9058133/97451332-90e2c180-192b-11eb-84af-8cb159736058.gif)

Reply:
The indicators look slick with the chevrons. And forget what I said about `Lerp`. You've accomplished it with `MINIMUM_NOTCH_DISTANCE`. And yes, please use this constant for calculating polygon points instead of having hardcoded numbers. Without doing that the connections between this value and the indicator aren't clear and it requires intimate knowledge to update this code in the future.